### ISSUE TITLE
doc about preStop hook is not correct

### ISSUE THREAD

**Issue by YuikoTakada** [kubernetes/website#35594]
**This is a Bug Report**

**Problem:**
In [document](https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/#discussion) written as:
```
Note: Kubernetes only sends the preStop event when a Pod is terminated. 
This means that the preStop hook is not invoked when the Pod is completed.
```
But actually, preStop hooks are also invoked when container in a pod is terminated due to LivenesProbe/StartupProbe failure. This is something different from "when a Pod is terminated".

**Proposed Solution:**
fix doc:
```
Kubernetes only sends the preStop event when a Pod or a container in the Pod is terminated.
This means that the preStop hook is not invoked when the Pod is completed.
```
**How can we reproduce it (as minimally and precisely as possible)?**
1. create yaml file "Liveness-preStop.yaml"
```
apiVersion: v1
kind: Pod
metadata:
  name: liveness-check2
spec:
  containers:
  - image: nginx
    name: nginx
    livenessProbe:
      httpGet:
        port: 80
        path: /
      failureThreshold: 5
      periodSeconds: 5
    lifecycle:
      preStop:
        exec:
          command:
            - sh
            - -c
            - |
              echo hook preStop. > /proc/1/fd/1
              date > /proc/1/fd/1
              exit 0
```
2. create a pod.
```
$ kubectl apply -f Liveness-preStop.yaml
```
3. make livenessProbe failure by deleting `/usr/share/nginx/html/index.html`
```
$ kubectl exec liveness-check2 -- rm /usr/share/nginx/html/index.html
```
4. check log
```
$ kubectl logs liveness-check2
...
$ kubectl logs liveness-check2 -f
2022/08/01 05:05:23 [error] 44#44: *80 directory index of "/usr/share/nginx/html/" is forbidden, client: 10.244.2.1, server: localhost, request: "GET / HTTP/1.1", host: "10.244.2.36:80"
10.244.2.1 - - [01/Aug/2022:05:05:23 +0000] "GET / HTTP/1.1" 403 153 "-" "kube-probe/1.24+" "-"
hook preStop.
Mon Aug  1 05:05:44 UTC 2022
2022/08/01 05:05:44 [notice] 1#1: signal 3 (SIGQUIT) received, shutting down
2022/08/01 05:05:44 [notice] 45#45: gracefully shutting down
```
-> preStop is invoked

**Page to Update:**
https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/#discussion

**Comment by dipankardas011** [comment]
@YuikoTakada If I understand the problem correctly then preStop hook is executed when Stop Event is sent to the container by the kubelet so if the preStop gets hanged or takes more time than graceTimeout it gets preStop hook failure event in pod logs

**Comment by dipankardas011** [comment]
I guess if it encounter an return code != 0 from the preStop then crashLoopBackOff

**Comment by YuikoTakada** [comment]
@dipankardas011 Thank you for your comments. Your explanation helped me understand. 
I think doc is not correct strictly. what written in doc is a little bit different from actual behavior(and also what you explained). I suggest fix doc in PR #35595 :
```
Kubernetes only sends the preStop event when a Pod is *terminated*.
```
```
Kubernetes only sends the preStop event when a Pod or a container in the Pod is *terminated*.
```

**Comment by dipankardas011** [comment]
@YuikoTakada I am just PR wrangler so you cannot make me to review pr sorry for that
